src/anemoi/training/diagnostics/callbacks/checkpoint.py

```diff
@@ -149,16 +149,22 @@ def _save_checkpoint(self, trainer: pl.Trainer, lightning_checkpoint_filepath: s
             tmp_metadata = model.metadata
             model.metadata = None
 
-            metadata = dict(**tmp_metadata)
+            tmp_supporting_arrays = model.supporting_arrays
```

Comment:
If `supporting_arrays` is also a dictionary couldn’t we just extend the metadata dict and write it there? Just thinking that doing so would remove the need for all this code, as it will be part of the metadata. Not sure if this has implications with the `anemoi-inference` interface or other packages so just a suggestion!

Reply:
No, because Numpy arrays are not JSON serialisable.